chia/util/augmented_chain.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, ClassVar, Dict, List, Optional, Set, Tuple, cast
+
+from chia.consensus.block_record import BlockRecord
+from chia.consensus.blockchain_interface import BlocksProtocol
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.types.full_block import FullBlock
+from chia.util.errors import Err
+from chia.util.ints import uint32
+
+
+def maybe_record(eb: Optional[Tuple[FullBlock, BlockRecord]]) -> Optional[BlockRecord]:
```

Comment:
It makes sense to make it private. I hesitate to make it a member though, since it doesn't depend on `self`. I'll rename it `_maybe_record()`

Reply:
my intention was that it would take a header hash and then it would use self to get the record from self._extra_blocks in my mind that would make it more readable and a bit cleaner